interview-questions.md

```diff
@@ -0,0 +1,58 @@
+# ASSESSMENT 2: Interview Practice Questions
+
+Answer the following questions.
+
+First, without external resources. Challenge yourself to answer from memory as if you were in a job interview.
+
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
+
+1. What is the difference between a parameter and an argument?
+
+Your answer: Parameters serve as place holders, until we call the function that takes the arguments value and substitute whats written in the parameter. 
+
+Researched answer: Note the difference between parameters and arguments: Function parameters are the names listed in the function's definition. Function arguments are the real values passed to the function. Parameters are initialized to the values of the arguments supplied.
+
+2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
+
+Your answer: element, index(optional), and array(optional)
+
+Researched answer: function() required, currentValue required, index optional, arr optional, thisValue optional
+
+3. What is the difference between map and filter?
+
+Your answer: .map transforms the elements in an array, while .filter selects specific elements from an array.
+
+Researched answer: Map: returns an array of pieces of information from the original array. In the callback function, return the data you wish to be part of the new array. Filter: returns a subset of the original array based on custom criteria.
```

Comment:
Not entirely correct on the explanation of map. It will return a new array with the same length as the original, but each element will be modified in the same way based on your logic.